librecad/src/lib/information/rs_information.cpp

```diff
@@ -109,12 +110,12 @@ bool RS_Information::isTrimmable(RS_Entity* e1, RS_Entity* e2) {
                     e1->getParent()==e2->getParent()) {
 
                 // in the same polyline
-                RS_EntityContainer* pl = e1->getParent();
+                RS_Polyline* pl = (RS_Polyline *)e1->getParent();
                 int idx1 = pl->findEntity(e1);
                 int idx2 = pl->findEntity(e2);
                 RS_DEBUG->print("RS_Information::isTrimmable: "
                                 "idx1: %d, idx2: %d", idx1, idx2);
-                if (abs(idx1-idx2)==1 || abs(idx1-idx2)==int(pl->count()-1)) {
+                if (abs(idx1-idx2)==1 || (pl->isClosed() && abs(idx1-idx2)==int(pl->count()-1))) {
```

Comment:
I think the assumption of sequential order makes sense in case of open polylines - logically, you might treat them as sequences of points (and radiuses, in cases of arcs), even though they're represented as lines and arcs internally. It shouldn't be possible to have 'gaps' within a polyline - the next line's start should always be the preceding line's end.

So, the existing iterator should guarantee looping through neighbours. :)

Whether that assumption is held consistently within the project is another question, but at least I haven't found any visible bugs in that area during normal use.